### PR TITLE
[lldb] Implement parsing of metadata generated by the JIT.

### PR DESCRIPTION
**Pull request by augusto2112**
TypeSystemSwiftTyperef requires metadata to be registered for it to be
able to do its work. This registers metadata generated by the JIT, which
are types declared in LLDB itself (including the REPL and playgrounds).

rdar://79075235

Depends on https://github.com/apple/swift/pull/38024 and https://github.com/apple/swift/pull/38029

### CODE REVIEW

File: lldb/include/lldb/Core/JITSection.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#ifndef LLDB_CORE_JIT_SECTION_H
```
Comment (adrian-prantl) [inline lldb/include/lldb/Core/JITSection.h:9]: *Here* a `-*- C++ -*-` marker would be nice :-)

File: lldb/include/lldb/Core/JITSection.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#define LLDB_CORE_JIT_SECTION_H
+
+#include "Section.h"
+
+namespace lldb_private {
+class JITSection : public Section {
```
Comment (adrian-prantl) [inline lldb/include/lldb/Core/JITSection.h:18]: Can you add a Doxygen comment to the class?

File: lldb/source/Expression/IRExecutionUnit.cpp [hunk 3]
```diff
@@ -336,6 +339,30 @@ void IRExecutionUnit::GetRunnableInfo(Status &error, lldb::addr_t &func_addr,
     m_execution_engine_up->setObjectCache(m_object_cache_up.get());
   }
 
+  // Record the size of the sections discounting the extra space used 
+  // by the JIT. This is a hack and should be removed when we switch
+  // from MCJIT to ORCJIT.
```
Comment (adrian-prantl) [inline lldb/source/Expression/IRExecutionUnit.cpp]: ///

Comment (augusto2112) [reply lldb/source/Expression/IRExecutionUnit.cpp]: This is declared in the middle of a function, do doxygen comments work here?

**Comment by augusto2112** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]

**Pull request merged**